src/main/scala/com/github/http4sbin/http/services/GetService.scala

```diff
@@ -0,0 +1,30 @@
+package com.github.http4sbin.http.services
+
+import org.http4s.HttpService
+import org.http4s.dsl._
+
+import argonaut._, Argonaut._
+
+import com.github.http4sbin.http.Http4sHelpers.{getHeadersAsJson, getQueryAsJson, getAbsoluteURI}
+
+object GetService {
+
+  val service = HttpService {
+    case req @ GET -> Root / "get" => {
+      val ip = req.remote match {
```

Comment:
So again, this is a case where HOF is better than pattern matching. Also, we can build stuff from previously defined constructs, rather than doing it manually:

`val ip = request.remote.map(_.getAddress.toString).asJson`

We have an `Option[A]`. We turn it into an `Option[B]` using `.map`. We already have a way to turn any `Option[A]` into `Json`, because an `EncodeJson` instance for it exists.